src/clj/game/cards/agendas.clj

```diff
@@ -1019,6 +1019,24 @@
               :effect (effect (gain :hand-size-modification 2))}
     :leave-play (effect (lose :hand-size-modification 2))}
 
+   "Remote Enforcement"
+   {:interactive (req true)
+    :optional {:prompt "Search R&D for a piece of ice to install protecting a remote server?"
+               :yes-ability {:delayed-completion true
+                             :prompt "Choose a piece of ice"
+                             :choices (req (filter ice? (:deck corp)))
+                             :effect (req (let [chosen-ice target]
+                                            (continue-ability state side
+                                                              {:delayed-completion true
+                                                               :prompt (str "Select a server to install " (:title chosen-ice) " on")
+                                                               :choices (filter #(not (#{"HQ" "Archives" "R&D"} %))
+                                                                                (corp-install-list state chosen-ice))
+                                                               :effect (effect
+                                                                        (corp-install chosen-ice target {:install-state :rezzed-no-rez-cost})
+                                                                        (shuffle! :deck)
+                                                                        (effect-completed eid))}
```

Comment:
You could do the `shuffle!` action first, then pass the `eid` to `corp-install`. Then there's no need for `effect-completed`. Unless there's some reason the shuffle would need to happen after the install (keep tracking of the card? I don't think that should be an issue, since the zone hasn't changed).